src/status_im/contexts/wallet/wallet_connect/events.cljs

```diff
@@ -50,13 +51,28 @@
               {:error error
                :event :wallet-connect/on-init-fail})))
 
+(defn- accounts-without-watched-with-customization-color
+  [all-accounts]
+  (->> all-accounts
+       (remove :watch-only?)
+       wallet-utils/accounts-with-customization-color))
+
 (rf/reg-event-fx
  :wallet-connect/on-session-proposal
  (fn [{:keys [db]} [proposal]]
-   (log/info "Received Wallet Connect session proposal: " {:id (:id proposal)})
-   {:db (assoc db :wallet-connect/current-proposal proposal)
-    :fx [[:dispatch
-          [:open-modal :screen/wallet.wallet-connect-session-proposal]]]}))
+   (let [all-accounts (get-in db [:wallet :accounts])
+         accounts     (accounts-without-watched-with-customization-color all-accounts)]
```

Comment:
Instead of duplicating the account information in the db, we could keep only the account address in the db and attach the account data in the subscription based on the address, since it's only needed for the UI (unless we need the entire account in the responding events I guess, but it doesn't seem to be the case as we only need the address there). Also, this way we also avoid the extraction of the `accounts-with-customization-color` subscription logic.